app/lib/theme/CustomDiffFields.tsx

```diff
@@ -20,193 +23,161 @@ const contentSuffixElement = (
   );
 };
 
-interface OverwriteNotificationArgument {
-  latestCommittedData: any;
-  isNumber?: boolean;
-  isDate?: boolean;
-  isMoney?: boolean;
-  isPercentage?: boolean;
-  numberOfDecimalPlaces?: number;
-}
+const renderData = (isDate: boolean | undefined, data: string | undefined) => {
+  if (isDate) {
+    return getLocaleFormattedDate(data);
+  }
+  return data;
+};
 
-const overwriteNotification = (
-  specs: OverwriteNotificationArgument
-): string | number | JSX.Element => {
-  const {
-    latestCommittedData,
-    isNumber,
-    isDate,
-    isMoney,
-    isPercentage,
-    numberOfDecimalPlaces,
-  } = specs;
+const renderTooltip = () => {
+  // TODO: implement this
+  return <></>;
+};
 
-  const displayString = "Latest committed value: ";
+const renderArrow = () => {
+  return (
+    <FontAwesomeIcon
+      className={"diff-arrow"}
+      size="lg"
+      color="black"
+      icon={faLongArrowAltRight}
+    />
+  );
+};
 
-  if (!latestCommittedData) {
-    return `${displayString}none`;
+const showStringDiff = (
+  id: string,
+  oldData: string | undefined,
+  newData: string | undefined,
+  latestCommittedData: string | undefined,
+  isDate: boolean,
+  contentSuffix?: string
+): JSX.Element => {
+  const [diffOldClsName, diffNewClsName, diffTextClsName] = [
+    "diffOld",
+    "diffNew",
+    "diffText",
+  ];
+
+  // Case 7 ->  111
+  if (oldData && newData && latestCommittedData) {
+    return (
+      <>
+        {latestCommittedData !== oldData && (
+          <>
+            <span
+              id={id && `${id}-${diffOldClsName}1`}
+              className={diffOldClsName}
+            >
+              {renderData(isDate, latestCommittedData)}
+            </span>
+            {contentSuffix && contentSuffixElement(id, contentSuffix)}
+            {renderArrow()}
+          </>
+        )}
+        <span id={id && `${id}-${diffOldClsName}`} className={diffOldClsName}>
+          {renderData(isDate, oldData)}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
+        {renderArrow()}
+        <span id={id && `${id}-${diffNewClsName}`} className={diffNewClsName}>
+          {renderData(isDate, newData)}
+        </span>
+        {contentSuffix && contentSuffixElement(id, contentSuffix)}
```

Comment:
This is pretty repetitive, maybe there could be functions for returning styled old, latestCommitted, and new components?

Reply:
I went down this path and found that some elements had embedded spans and made it hard to style.